backend/modules/notifications/routers/student_notifications.py

```
@@ -0,0 +1,167 @@
+        )
+    
+    return {"message": "All notifications marked as read"}
+
+
+# ==================== TOKEN-BASED INVITATION ENDPOINTS ====================
+# These endpoints don't require authentication and are used from email links
+
+@router.get("/invitations/{invitation_token}")
+async def get_invitation_by_token(
+    invitation_token: str,
+    service: NotificationService = Depends(get_notification_service)
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)